internal/services/oracle/oracle_exascale_database_storage_vault_resource.go

```diff
@@ -0,0 +1,289 @@
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (ExascaleDatabaseStorageVaultResource) ModelObject() interface{} {
+	return &ExascaleDatabaseStorageVaultResource{}
+}
+
+func (ExascaleDatabaseStorageVaultResource) ResourceType() string {
+	return "azurerm_oracle_exascale_database_storage_vault"
+}
+
+func (r ExascaleDatabaseStorageVaultResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 120 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Oracle.OracleClient.ExascaleDbStorageVaults
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var model ExascaleDatabaseStorageVaultResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
+
+			id := exascaledbstoragevaults.NewExascaleDbStorageVaultID(subscriptionId,
+				model.ResourceGroupName,
+				model.Name)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			param := exascaledbstoragevaults.ExascaleDbStorageVault{
+				Name:     pointer.To(model.Name),
+				Location: location.Normalize(model.Location),
+				Zones:    pointer.To(model.Zones),
+				Tags:     pointer.To(model.Tags),
+				Properties: &exascaledbstoragevaults.ExascaleDbStorageVaultProperties{
+					DisplayName:                   model.DisplayName,
+					AdditionalFlashCacheInPercent: pointer.To(model.AdditionalFlashCachePercentage),
+					Description:                   pointer.To(model.Description),
+					TimeZone:                      pointer.To(model.TimeZone),
+					HighCapacityDatabaseStorageInput: exascaledbstoragevaults.ExascaleDbStorageInputDetails{
+						TotalSizeInGbs: model.HighCapacityDatabaseStorage[0].TotalSizeInGb,
+					},
+				},
+			}
+
+			if err := client.CreateThenPoll(ctx, id, param); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r ExascaleDatabaseStorageVaultResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Oracle.OracleClient.ExascaleDbStorageVaults
+			id, err := exascaledbstoragevaults.ParseExascaleDbStorageVaultID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var model ExascaleDatabaseStorageVaultResourceModel
+			if err = metadata.Decode(&model); err != nil {
+				return err
+			}
+
+			_, err = client.Get(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			update := &exascaledbstoragevaults.ExascaleDbStorageVaultTagsUpdate{}
+			update.Tags = pointer.To(model.Tags)
```

Comment:
We can just instantiate it in one go without doing a separate access later on

```suggestion
			update := &exascaledbstoragevaults.ExascaleDbStorageVaultTagsUpdate{
				Tags: pointer.To(model.Tags),
			}
```